src/crypto/equihash.cpp

```diff
@@ -514,8 +639,12 @@ bool Equihash<N,K>::OptimisedSolve(const eh_HashState& base_state,
             icv.reserve(recreate_size);
             for (eh_index j = 0; j < recreate_size; j++) {
                 eh_index newIndex { UntruncateIndex(partialSoln.get()[i], j, CollisionBitLength + 1) };
-                icv.emplace_back(base_state, newIndex, HashLength,
-                                 CollisionBitLength, CollisionByteLength);
+                if (j == 0 || newIndex % IndicesPerHashOutput == 0) {
```

Comment:
I can't convince myself this is correct. The newIndex values are not necessarily contiguous, right?